cellfinder_napari/train/train.py

```diff
@@ -0,0 +1,184 @@
+        save_weights : bool
+            Only store the model weights, and not the full model
+            Useful to save storage space
+        save_checkpoints : bool
+            Store the model at intermediate points during training
+        save_progress : bool
+            Save training progress to a .csv file
+        epochs : int
+            Number of training epochs
+            (How many times to use each training data point)
+        learning_rate : float
+            Learning rate for training the model
+        batch_size : int
+            Training batch size
+        test_fraction : float
+            Fraction of training data to use for validation
+        number_of_free_cpus : int
+            How many CPU cores to leave free
+        reset_button : PushButton
+            Reset parameters to default
+        """
+
+        trained_model = None if trained_model == Path.home() else trained_model
+        model_weights = None if model_weights == Path.home() else model_weights
+
+        training_data_inputs = TrainingDataInputs(yaml_files, output_directory)
+
+        optional_network_inputs = OptionalNetworkInputs(
+            trained_model,
+            model_weights,
+            model_depth,
+            pretrained_model,
+        )
+
+        optional_training_inputs = OptionalTrainingInputs(
+            continue_training,
+            augment,
+            tensorboard,
+            save_weights,
+            save_checkpoints,
+            save_progress,
+            epochs,
+            learning_rate,
+            batch_size,
+            test_fraction,
+        )
+
+        misc_training_inputs = MiscTrainingInputs(number_of_free_cpus)
+
+        if yaml_files[0] == Path.home():  # type: ignore
+            show_info("Please select a YAML file for training")
+        else:
+            worker = run_training(
+                training_data_inputs,
+                optional_network_inputs,
+                optional_training_inputs,
+                misc_training_inputs,
+            )
+            worker.start()
+
+    widget.header.value = widget_header
+    widget.header.native.setOpenExternalLinks(True)
+
+    @widget.reset_button.changed.connect
+    def restore_defaults():
+        defaults = {
+            **TrainingDataInputs.defaults(),
+            **OptionalNetworkInputs.defaults(),
+            **OptionalTrainingInputs.defaults(),
+            **MiscTrainingInputs.defaults(),
+        }
+        for name, value in defaults.items():
+            # ignore fields with no default
+            if value is not None:
+                getattr(widget, name).value = value
+
+    return widget
+
+
+@thread_worker
+def run_training(
+    training_data_inputs: TrainingDataInputs,
+    optional_network_inputs: OptionalNetworkInputs,
+    optional_training_inputs: OptionalTrainingInputs,
+    misc_training_inputs: MiscTrainingInputs,
+):
+    from cellfinder_core.train.train_yml import run as train_yml
```

Comment:
Why is this being imported here and not at the top?

Reply:
Good point. Moved.